test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java

```diff
@@ -593,27 +593,39 @@ private void installWhiteBox() {
      * Disable IR verification completely in certain cases.
      */
     private void disableIRVerificationIfNotFeasible() {
+        if (!irVerificationPossible) {
+            return;
+        }
+
+        boolean debugTest, irTest, nonWhiteListedTest;
+
+        debugTest = Platform.isDebugBuild() && !Platform.isInt() && !Platform.isComp();
+        irTest = hasIRAnnotations();
+        // No IR verification is done if additional non-whitelisted JTreg VM or Javaoptions flag is specified.
+        List<String> nonWhiteListedFlags = anyNonWhitelistedJTregVMAndJavaOptsFlags();
+        nonWhiteListedTest = nonWhiteListedFlags.isEmpty();
+
+        irVerificationPossible = debugTest && irTest && nonWhiteListedTest;
         if (irVerificationPossible) {
-            irVerificationPossible = Platform.isDebugBuild() && !Platform.isInt() && !Platform.isComp();
-            if (!irVerificationPossible) {
-                System.out.println("IR verification disabled due to not running a debug build (required for PrintIdeal" +
-                                   "and PrintOptoAssembly), running with -Xint, or -Xcomp (use warm-up of 0 instead)");
-                return;
-            }
+            return;
+        }
 
-            irVerificationPossible = hasIRAnnotations();
-            if (!irVerificationPossible) {
-                System.out.println("IR verification disabled due to test " + testClass + " not specifying any @IR annotations");
-                return;
-            }
+        System.out.println("IR verification disabled due to:");
+        if (!debugTest) {
+            System.out.println("\tnot running a debug build (required for PrintIdeal and PrintOptoAssembly), " +
```

Comment:
I like the improvements so far. But I think we should not use tabs since the size could be specific to the machine we are running on. I suggest to use `-` instead which could use the following structure:
```
IR verification disabled due to the following reason(s):
- Reason 1
- Reason 2
...
- Using non-whitelisted JTreg VM or Javaoptions flag(s):
  - Non-whitelisted flag 1
  - Non-whitelisted flag 2
  - ...
```

Reply:
Make sense to me, fixed.